### PR TITLE
pcsc-safenet: 10.8.28 -> 10.8.1050

### PR DESCRIPTION
**Pull request by tmarkov**
## Description of changes

Update Safenet Authentication Client to 10.8.1050 (or 10.8 R1).

Switch to a supported, up to date openssl version.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
  - [X] Tested SACTools, and the pkcs11 module library.
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

@wldhx 

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by tmarkov** [comment]
Actually, should probably still use digicert, because globasign seem to just overwrite the same file when updating, and we don't want that.

File: pkgs/tools/security/pcsc-safenet/default.nix [hunk 1]
```diff
@@ -4,23 +4,21 @@
...
-    let
-      versionWithUnderscores = builtins.replaceStrings ["."] ["_"] version;
-    in fetchzip {
-      url = "https://www.digicert.com/StaticFiles/SAC_${versionWithUnderscores}_GA_Build.zip";
-      hash = "sha256-7XWj3T9/KnmgQ05urOJV6dqgkAS/A2G7efnqjQO2ing=";
+  src = fetchzip {
+      # Version name in archive different from deb file inside
+      url = "https://www.digicert.com/StaticFiles/Linux_SAC_10_8_R1_GA.zip";
+      hash = "sha256-Wh2Ax4ZVFKqn0yDwZmwvtUqwQNYyBng08IPfemHzZC0=";
     };
```
Comment (vlaci) [inline pkgs/tools/security/pcsc-safenet/default.nix:22]:
```suggestion
  };
```

**Review by vlaci — CHANGES_REQUESTED** [review]
result of `nixpkgs-review`:
```
1 package built:
pcsc-safenet
```

I had one stylistic nitpick, otherwise LGTM

**Review by vlaci — APPROVED** [review]

**Review by FliegendeWurst — APPROVED** [review]

**Comment by KiaraGrouwstra** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 341808`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>pcsc-safenet</li>
  </ul>
</details>

File: pkgs/tools/security/pcsc-safenet/default.nix
Comment (FliegendeWurst) [inline pkgs/tools/security/pcsc-safenet/default.nix:1]: Please rebase, this package has been moved to pkgs/by-name

#354531

Comment (tmarkov) [reply pkgs/tools/security/pcsc-safenet/default.nix:1]: Done

File: pkgs/by-name/pc/pcsc-safenet/package.nix [hunk 2]
```diff
@@ -4,23 +4,20 @@
...
 
   # extract debian package from larger zip file
-  src =
-    let
-      versionWithUnderscores = builtins.replaceStrings ["."] ["_"] version;
-    in fetchzip {
-      url = "https://www.digicert.com/StaticFiles/SAC_${versionWithUnderscores}_GA_Build.zip";
-      hash = "sha256-7XWj3T9/KnmgQ05urOJV6dqgkAS/A2G7efnqjQO2ing=";
+  src = fetchzip {
+      url = "https://www.globalsign.com/en/safenet-drivers/USB/10.8/GlobalSign-SAC-Ubuntu-2204.zip";
```
Comment (FliegendeWurst) [inline pkgs/by-name/pc/pcsc-safenet/package.nix:19]: This URL does not include the full version. Therefore it is most likely not suitable.

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Pull request closed without merging**